### PR TITLE
workflow fix+pinot metrics

### PR DESCRIPTION
**Pull request by Harnoor-se7en**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java [hunk 1]
```diff
@@ -429,6 +447,46 @@ public Observable<Row> handleRequest(
...
+    }
+  }
+
+  private void measureRequestAge(ExecutionContext executionContext) {
+    try {
+      Preconditions.checkNotNull(executionContext);
+      Preconditions.checkNotNull(executionContext.getQueryTimeRange());
+      if (executionContext.getQueryTimeRange().isPresent()) {
+        Optional<QueryTimeRange> queryAge = executionContext.getQueryTimeRange();
+        if (queryAge.isPresent()) {
```
Comment (suddendust) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java]: I think this condition will always be true. So, you can remove this `if` and put an `assert queryRange.isPresent()`.

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java [hunk 2]
```diff
@@ -429,6 +447,46 @@ public Observable<Row> handleRequest(
...
+          Instant startTimeInstant = (queryAge.get().getStartTime());
```
Comment (suddendust) [inline query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java]: _nit_: No need to put them into separate variables? Just inline them.

**Review by suddendust — CHANGES_REQUESTED** [review]
Minor comments.

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java (hunk 1 above)
Comment (Harnoor-se7en) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java]: isn't assert only meant for tests? In codebase also assert is being used only in tests.

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java (hunk 2 above)
Comment (Harnoor-se7en) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java]: Sure

File: query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java (hunk 1 above)
Comment (Harnoor-se7en) [reply query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java]: Done

**Review by suddendust — APPROVED** [review]
LGTM

**Pull request merged**